### PR TITLE
Upgrade module to use AWS SDK v3

### PR DESCRIPTION
**Pull request by halkyon**

### CODE REVIEW

**Comment by Firesphere** [comment]
Robots talking to each other. Is the Singularity there?

File: code/model/DynamoDbSession.php [hunk 1]
```diff
@@ -40,14 +42,20 @@ public static function get()
...
-                $dynamoOptions['secret'] = AWS_SECRET_KEY;
+                $dynamoOptions['credentials']['key'] = AWS_ACCESS_KEY;
+                $dynamoOptions['credentials']['secret'] = AWS_SECRET_KEY;
             } else {
                 // cache credentials when IAM fetches the credentials from EC2 metadata service
-                // this will use doctrine/cache (included via composer) to do the actual caching into the filesystem
+                // this will use doctrine/cache (included via composer) to do the actual caching into APCu
                 // http://docs.aws.amazon.com/aws-sdk-php/guide/latest/performance.html#cache-instance-profile-credentials
-                $dynamoOptions['credentials.cache'] = true;
+                $dynamoOptions['credentials'] = new DoctrineCacheAdapter(new ApcuCache());
```
Comment (stojg) [inline code/model/DynamoDbSession.php:58]: This makes this module to require apcu?

Comment (halkyon) [inline code/model/DynamoDbSession.php:58]: yeah it does, should we add the composer requirement? Although it's only APCu on instance profiles.

Comment (stojg) [inline code/model/DynamoDbSession.php:58]: Is it possible to fallback to PhpFileCache or ArrayCache instead of require the extension?

Comment (stojg) [inline code/model/DynamoDbSession.php:58]: e.g

```
if function_exists('apcu_fetch') { .... 
```

Comment (halkyon) [inline code/model/DynamoDbSession.php:58]: I'll have a look, will try to pull across what `credentials.cache` did in AWS SDK v2.

Comment (halkyon) [inline code/model/DynamoDbSession.php:58]: @stojg Actually, this doesn't change any requirements from the original SDK which used APC: https://github.com/aws/aws-sdk-php/blob/2.8/src/Aws/Common/Credentials/Credentials.php#L318

**Pull request merged**